app/services/record_declaration.rb

```diff
@@ -111,16 +111,29 @@ def validate_milestone_exists
   end
 
   def participant_declaration
-    @participant_declaration ||= participant_profile
-                                   .participant_declarations
-                                   .submitted
-                                   .or(
-                                     participant_profile
-                                       .participant_declarations
-                                       .billable,
-                                   )
-                                   .find_or_create_by!(declaration_parameters)
-                                   .tap { |pd| pd.update!(uplift_flags) }
+    @participant_declaration ||= find_participant_declaration.tap { |pd| pd.update!(uplift_flags) }
+  end
+
+  def find_participant_declaration
+    participant_profile
+      .participant_declarations
+      .submitted
+      .or(
+        participant_profile
+          .participant_declarations
+          .billable,
+      )
+      .find_or_create_by!(declaration_parameters)
+  rescue ActiveRecord::RecordNotUnique
+    participant_profile
+      .participant_declarations
+      .submitted
+      .or(
+        participant_profile
+          .participant_declarations
+          .billable,
+      )
+      .find_by!(declaration_parameters)
```

Comment:
I'm not sure I see the need for the query to be repeated after the `rescue`. If the initial `find_or_create_by!` doesn't find a match the `find_by!` can't either. Should we just `fail`/`raise` instead?

Reply:
@peteryates that was to manage a race condition where it appears the a lead provider was able to double post a declaration. 
I've actually replaced that with `create_or_find_by!` which actually does  the `ActiveRecord::RecordNotUnique` exception handling and issue the find.